lib/workers/repository/finalize/index.ts

```diff
@@ -18,9 +18,9 @@ export async function finalizeRepo(
   repoConfig: AllConfig,
 ): Promise<void> {
   await checkReconfigureBranch(config, repoConfig);
-  await repositoryCache.saveCache();
   await pruneStaleBranches(config, branchList);
   await ensureIssuesClosing();
+  await repositoryCache.saveCache();
```

Comment:
It's better to make the minimal change to avoid introducing other cache-related errors

```suggestion
  await pruneStaleBranches(config, branchList);
  await repositoryCache.saveCache();
  await ensureIssuesClosing();
```